### ISSUE TITLE
Copy `spatial_partitions` when copying `GeoDataFrame`

### ISSUE THREAD

**Issue by DahnJ** [geopandas/dask-geopandas#55]
I would expect the `spatial_partitions` to be copied when using `df.copy()`.

```python
import geopandas
import dask
import dask_geopandas
dask.config.set(scheduler='single-threaded')

df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))
ddf = dask_geopandas.from_geopandas(df, npartitions=4)
ddf.calculate_spatial_partitions()
ddfc = ddf.copy()
ddfc.spatial_partitions  # is None
```

Is there a reason to not copy spatial partitions that I didn't think of?

I thought about how this could work and didn't think of anything better than overriding dask's `copy` method. Something like

```python
def copy(self):
    self_copy = super().copy()
    self_copy.spatial_partitions = self.spatial_partitions
    return self_copy
```

It would be great if `spatial_partitions` could get copied automatically, as `crs` for example does. However, `crs` is actually a property of `_meta`, which does get automatically copied by dask. Not sure how spatial partitions could be attached to the `_meta`, or if there's another mechanism that could act similarly.

If I'm headed in the right direction, then I'm happy to unit test & PR.

**Comment by jorisvandenbossche** [comment]
Yes, I agree that would certainly be expected to be propagated to the copy.

I suspect there will be other methods as well where we need to fix this.

On the short term, I think your proposed fix of overriding the method like
that is fine (maybe just copy the actual spatial_partitions as well). PR
certainly welcome!
Long term it would be good to have a nicer way to have the spatial
paritions copied over, but probably something to discuss with the dask team
what this could look like.

Op za 12 jun. 2021 12:10 schreef Dahn ***@***.***>:

**Comment by martinfleis** [comment]
Closed by #56
